src/query/server/cost_reporters.go

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package server
+
+import (
+	"sync"
+
+	"github.com/m3db/m3/src/cmd/services/m3query/config"
+	qcost "github.com/m3db/m3/src/query/cost"
+	"github.com/m3db/m3/src/x/cost"
+	"github.com/m3db/m3x/instrument"
+
+	"github.com/uber-go/tally"
+)
+
+// Reporters for our query/cost.ChainedEnforcer instances.
+
+const (
+	queriesOverLimitMetric  = "over_datapoints_limit"
+	datapointsMetric        = "datapoints"
+	datapointsCounterMetric = "datapoints_counter"
+	maxDatapointsHistMetric = "max_datapoints_hist"
+)
+
+// newConfiguredChainedEnforcer returns a ChainedEnforcer with 3 configured
```

Comment:
Nice comment! Might be worth extracting/duplicating some of this in docs since there's not a lot of visibility into cost accounting at the moment?

Reply:
Probably in a query howto? We should sync with Ben about this as well, he's been handling a lot of documentation